pkg/tsdb/cloudwatch/models/request_parser.go

```diff
@@ -42,52 +45,52 @@ type QueryJson struct {
 	Alias             string                 `json:"alias,omitempty"`
 }
 
-// parseQueries parses the json queries and returns a map of cloudWatchQueries by region. The cloudWatchQuery has a 1 to 1 mapping to a query editor row
-func parseQueries(queries []backend.DataQuery, startTime time.Time, endTime time.Time, dynamicLabelsEnabled bool) (map[string][]*cloudWatchQuery, error) {
-	requestQueries := make(map[string][]*cloudWatchQuery)
+// ParseQueries parses the json queries and returns a map of cloudWatchQueries by region. The cloudWatchQuery has a 1 to 1 mapping to a query editor row
+func ParseQueries(queries []backend.DataQuery, startTime time.Time, endTime time.Time, dynamicLabelsEnabled bool) (map[string][]*CloudWatchQuery, error) {
```

Comment:
I'd like to think that parsing a dataquery json into a dto is something that is done inside the model. So in cloudwatch_query you'd have a `ParseCloudWatchQuery(backend.DataQuery) (*CloudWatchQuery, error)`. Then the business logic (a future service) would be responsible for splitting the queries by region. But maybe this is something you're planning to do in the next step of this refactoring?

Reply:
Hmmm I didn't remember to separate the grouping by region out of `ParseQueries`.

Do you think that `ParseQueries` still parses a slice of data queries `[]backend.DataQuery` and returns a slice of `[]*CloudWatchQuery`?

The next PRs I'd like to prepare will add tests where they're missing or shift them to test at the `ParseQueries` level. I ought to be able to shift region grouping tests out of that.

Reply:
Sorry I may have forgotten to mention this before. But I think we should start with iterating through all data queries in the handler and parse them into DTO's (CloudWatchQueries). Then all handlers (routes or query data handlers) share the same pattern across the whole CloudWatch backend. Even though it will cost slightly more from a performance view to split by region later, it will still be worth it IMO. If you want to do this in another PR that's totally fine with me.

Reply:
Thanks, I'll go ahead and merge this then. 
Summary of what we discussed over the call:
* shift out the regions grouping (as stated above)
* ParseQueries will be in `models`. 
* ParseQueries will take an array of backend.DataQuery and return the slice of CloudWatchQuery (DTO)
* ParseQueries will decode JSON, validate and set defaults, perform migrations resulting in the CloudWatchQuery
* leaving that executeTimeSeriesQuery "handler" alone for now